01/1C.hector_hist_co2.Rmd

```diff
@@ -0,0 +1,101 @@
+---
+title: "Hector [CO2] vs. Observations"
+output: html_document
+author: "Peter Scully"
+date: "2024-05-31"
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+## Getting the data
+
+First, let's import the relevant libraries and setup our directory location
+```{r import_and_loc}
+library(hector)
+library(ggplot2)
+library(dplyr)
+
+INPUT_DIR <- file.path(here::here(), '01')
+```
+
+Now, let's read the historical data into a dataframe
+
+```{r hist_table}
+hist_data <- read.table(file.path(INPUT_DIR, 'co2_annmean_mlo.txt'), 
+                        skip = 43, 
+                        col.names = c("year", "mean", "unc"))
+hist_data
+```
+
+We also need to get the default Hector data:
+
+```{r hector_run}
+# Setting the desired date range and variables
+yrs <- 1959:2023
+hector_vars <- CONCENTRATIONS_CO2()
+
+# Running the default ssp245 run
+ini_file <- system.file("input/hector_ssp245.ini", package = "hector")
+core <- newcore(ini_file)
+run(core)
+hector_data <- fetchvars(core, yrs, vars = hector_vars)
+
+hector_data
+
+```
+
+## Making the comparison plot
+
+Now that we have both the historical and model data, we need to combine the
+data frames so we can make a comparison plot:
+```{r combining data frames}
+
+# First, let's delete the unnecessary columns
+hist_data <- subset(hist_data, select = c('year', 'mean'))
+hector_data <- subset(hector_data, select = c('year', 'value'))
+
+# Now, let's make the column names match so we can combine the data frames
+colnames(hist_data) <- c('year', 'CO2_concentration')
+colnames(hector_data) <- c('year', 'CO2_concentration')
+
+# Finally, let's add a column to distinguish which data comes from which source
+hist_data$data_source <- "Historical Record"
+hector_data$data_source <- "Hector"
+
+# Combining the data frames
+combined_data <- rbind(hist_data, hector_data)
+
+```
+
+With this new data frame, we can now make the actual comparison plot
+
+```{r plotting}
```

Comment:
@ptrscll could you add the uncertainty range to the [CO2] obs? (hint: use geom_ribbon)